GetProfile/handler.ts

```
@@ -42,14 +52,53 @@ type IGetProfileHandler = (
   fiscalCode: FiscalCode
 ) => Promise<IGetProfileHandlerResult>;
 
+export const withIsEmailAlreadyTaken = (
+  profileEmailReader: IProfileEmailReader,
+  uniqueEmailEnforcementEnabled: boolean
+) => (profile: ExtendedProfile): T.Task<ExtendedProfile> =>
+  pipe(
+    TE.of(profile),
+    // VALID ONLY IF FF_UNIQUE_EMAIL_ENFORCEMENT IS ENABLED
+    // Check if the e-mail address associated with the retrived
+    // profile was validated. If was not validated, continue with
+    // uniqueness checks.
+    TE.filterOrElse(
+      ({ is_email_validated }) =>
+        !is_email_validated && uniqueEmailEnforcementEnabled,
+      () => true
+    ),
+    TE.chain(({ email }) =>
+      pipe(
+        // Check if the e-mail is already taken (returns a boolean).
+        // If there are problems checking the uniqueness of the provided
+        // e-mail address, assume that the e-mail is not unique (already taken).
+        // isEmailAlreadyTakenTE(profileEmailReader, profile.email),
+        TE.tryCatch(
+          () =>
+            isEmailAlreadyTaken(email)({
+              profileEmails: profileEmailReader
+            }),
+          () => false
```

why not return an error instead?